mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,178 @@
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        :param ss: SparkContext 
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data =  df.where( ( ( len(df.alternateAllele ) > 1 )) & (  (df.alleles[0] == u'ALT') | (df.alleles[1] == u'ALT'))).groupBy("sampleid").count().collect()
+        counts = [ sample.asDict()['count'] for sample in data ]
+        self.counts = counts
+    def plot(self):
+        plt.hist(self.counts,bins=default_bins)
+        plt.title("Variants per Sample")
+        plt.xlabel("Variants")
+        plt.ylabel("Samples")
+        plt.show()
+
+
+class HetHomRatioPerSample(object):
+    """
+    Plot Histogram of Het/Hom ratio per sample
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        :param ss: SparkContext 
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data_het =  df.where( (df.alleles[0] == u'REF') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        hets =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_het }
+        data_hom =  df.where( (df.alleles[0] == u'ALT') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        homs =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_hom }
+        self.het_hom_ratio = [ float(hets[sampleid])/float(homs[sampleid]) for sampleid in homs.keys() ]
+
+    def plot(self, testMode = False):
+        if (not testMode):
+          plt.hist(self.het_hom_ratio,bins=default_bins)
+          plt.title("Histogram of Het/Hom Ratio")
+          plt.xlabel("Het/Hom Ratio")
+          plt.ylabel("Sample Count")
+          plt.show()
+        return self.het_hom_ratio
+
+
+class CallRatePerSample(object):
+    """
+    Plot Histogram of call rate per sample
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        :param ss: SparkContext 
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data_called =  df.where( (df.alleles[0] != u'NO_CALL')).groupBy("sampleid").count().collect()
+        called =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_called }
+        data_missing =  df.where( (df.alleles[0] == u'NO_CALL')).groupBy("sampleid").count().collect()
+        missing =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_missing }
+        self.call_rate = [ float(called[sampleid])/(float(missing[sampleid]+float(called[sampleid]))) for sampleid in called.keys() ]
+
+    def plot(self, testMode = False):
+        if (not testMode):
+          plt.hist(self.call_rate,bins=default_bins)
+          plt.title("Histogram of Call Rate per sample")
+          plt.xlabel("Call Rate")
+          plt.ylabel("Sample Count")
+          plt.show()
+        return self.call_rate
+
+class QDDist(object):
```

Comment:
There are a lot of random class names. Please look at https://github.com/bigdatagenomics/mango/blob/master/mango-python/bdgenomics/mango/alignments.py for an idea of how to better group/divide functionality by classes